virtualizarr/tests/test_xarray.py

```diff
@@ -375,10 +378,48 @@ def test_read_from_url(self, filetype, url):
         if filetype in ["grib", "jpg", "hdf4"]:
             with pytest.raises(NotImplementedError):
                 vds = open_virtual_dataset(url, reader_options={}, indexes={})
+        elif filetype == "hdf5":
+            vds = open_virtual_dataset(
+                url,
+                group="science/LSAR/GCOV/grids/frequencyA",
+                drop_variables=["listOfCovarianceTerms", "listOfPolarizations"],
+                indexes={},
+                reader_options={},
+            )
+            assert isinstance(vds, xr.Dataset)
         else:
             vds = open_virtual_dataset(url, reader_options={}, indexes={})
             assert isinstance(vds, xr.Dataset)
 
+    def test_virtualizarr_vs_local_nisar(self):
+        # Open group directly from locally cached file with xarray
+        url = "https://nisar.asf.earthdatacloud.nasa.gov/NISAR-SAMPLE-DATA/GCOV/ALOS1_Rosamond_20081012/NISAR_L2_PR_GCOV_001_005_A_219_4020_SHNA_A_20081012T060910_20081012T060926_P01101_F_N_J_001.h5"
+        tmpfile = fsspec.open_local(
+            f"filecache::{url}", filecache=dict(cache_storage="/tmp", same_names=True)
+        )
+        hdf_group = "science/LSAR/GCOV/grids/frequencyA"
+        dsXR = xr.open_dataset(
+            tmpfile,
+            engine="h5netcdf",
+            group=hdf_group,
+            drop_variables=["listOfCovarianceTerms", "listOfPolarizations"],
+            phony_dims="access",
+        )
+
+        # save group reference file via virtualizarr, then open with engine="kerchunk"
+        vds = open_virtual_dataset(
+            tmpfile,
+            group=hdf_group,
+            indexes={},
+            drop_variables=["listOfCovarianceTerms", "listOfPolarizations"],
+        )
+        tmpref = "/tmp/cmip6.json"
+        vds.virtualize.to_kerchunk(tmpref, format="json")
+        dsV = xr.open_dataset(tmpref, engine="kerchunk")
+
+        # xrt.assert_identical(dsXR, dsV) #Attribute order changes
```

Comment:
Attribute order changes? But xarray's `.attrs` is a dict, which shouldn't require order of keys to be the same?

Reply:
Sorry for the lingering misleading notes! It seems the problem is attribute values containing numpy arrays (full_ds) versus python objects (virtual_ds):

```
In [33]: full_ds.air.attrs
Out[33]: 
{'long_name': '4xDaily Air temperature at sigma level 995',
 'units': 'degK',
 'precision': np.int16(2),
 'GRIB_id': np.int16(11),
 'GRIB_name': 'TMP',
 'actual_range': array([185.16, 322.1 ], dtype=float32)}
...

In [34]: vds.air.attrs
Out[34]: 
{'GRIB_id': 11,
 'GRIB_name': 'TMP',
 'actual_range': [185.16000366210938, 322.1000061035156],
 'long_name': '4xDaily Air temperature at sigma level 995',
 'parent_stat': 'Other',
 'precision': 2,
 'scale_factor': 0.01,
...
```

Reply:
Okay this sounds like an xarray assertion issue separate from this PR, so especially if we can reproduce it maybe we can just leave it for later?

Reply:
Yep let's keep it separate - I think just reopen https://github.com/zarr-developers/VirtualiZarr/issues/161 for now. I added comments there to easily reproduce

Reply:
I think the error I was talking about here is separate from both #161 and this PR.